### DISCUSSION TITLE
E-commerce Product Management System using json-server, react, and axios

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#187] [Tasks]
# Part-1

Let's set up a more comprehensive example that showcases an e-commerce product management system using `json-server`, `react`, and `axios` for data fetching.

### 1. Set up a new React project using Create React App (CRA):

Reasoning: CRA is an officially supported way to create single-page React applications. It offers a modern build setup with no configuration.

```bash
npx create-react-app product-manager
cd product-manager
```

### 2. Install required dependencies:

```bash
npm install axios json-server
```

- **axios**: A popular HTTP client for making asynchronous requests.
- **json-server**: Mock server for REST API.

### 3. Create a mock database:

Create a `db.json` at the root:

```json
{
  "products": [
    { "id": 1, "name": "Laptop", "price": 1000, "description": "A high performance laptop" },
    { "id": 2, "name": "Mouse", "price": 20, "description": "Ergonomic wireless mouse" }
  ]
}
```

Reasoning: This file acts as your mock database. You can list products, their prices, and descriptions.

### 4. Setting up `json-server`:

Modify the `package.json` scripts to run both React and json-server:

```json
"scripts": {
  "start": "react-scripts start",
  "server": "json-server --watch db.json --port 3001",
  "dev": "concurrently --kill-others \"npm run start\" \"npm run server\""
}
```

Reasoning: This setup allows you to run both servers concurrently. The React app will run on its default port 3000, while JSON Server will run on port 3001.

### 5. Fetch products in React using axios:

Create a component `Products.js`:

```jsx
import React, { useState, useEffect } from 'react';
import axios from 'axios';

function Products() {
  const [products, setProducts] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    axios.get('http://localhost:3001/products')
      .then(response => {
        setProducts(response.data);
      })
      .catch(error => {
        setError(error);
      });
  }, []);

  if (error) return <div>Error fetching products!</div>;

  return (
    <div>
      {products.map(product => (
        <div key={product.id}>
          <h3>{product.name} - ${product.price}</h3>
          <p>{product.description}</p>
        </div>
      ))}
    </div>
  );
}

export default Products;
```

Reasoning:

- **axios.get**: Axios uses promises for handling asynchronous operations, and here we're fetching data from our mock server.
- **useState**: A React hook to declare state in functional components.
- **useEffect**: A React hook to perform side effects, in this case, data fetching.

### 6. Display the `Products` component:

In `App.js`:

```jsx
import React from 'react';
import './App.css';
import Products from './Products';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Product Manager</h1>
      </header>
      <Products />
    </div>
  );
}

export default App;
```

Reasoning: The main application component imports and renders the `Products` component.

### 7. Running the application:

```bash
npm run dev
```

Reasoning: This command concurrently starts both the React development server and the `json-server`, allowing you to develop and test your React app against the mock backend seamlessly.

This comprehensive exercise showcases how you can integrate `json-server` into a React project, use `axios` for fetching data, and understand the importance of each step. The steps cover the basics of setting up a mock backend, fetching data in a React app, and handling possible errors.

# Part-2

Alright, let's continue expanding our project. This time, we're aiming to enhance the user experience and incorporate more complex features.

### 8. Installing Bootstrap for richer UI:

```bash
npm install bootstrap
```

In `src/index.js`, import the Bootstrap CSS:

```javascript
import 'bootstrap/dist/css/bootstrap.min.css';
```

### 9. Create a Product Detail Page:

This allows users to click on a product to get more details.

`ProductDetail.js`:

```jsx
import React, { useState, useEffect } from 'react';
import axios from 'axios';

function ProductDetail({ match }) {
    const [product, setProduct] = useState({});
    const [error, setError] = useState(null);

    useEffect(() => {
        axios.get(`http://localhost:3001/products/${match.params.id}`)
            .then(response => {
                setProduct(response.data);
            })
            .catch(error => {
                setError(error);
            });
    }, [match.params.id]);

    if (error) return <div>Error fetching product details!</div>;

    return (
        <div className="container mt-4">
            <h2>{product.name}</h2>
            <h4>${product.price}</h4>
            <p>{product.description}</p>
        </div>
    );
}

export default ProductDetail;
```

### 10. Add Bootstrap classes to enhance UI:

Modify the `Products.js` to utilize Bootstrap classes and create cards:

```jsx
return (
    <div className="container mt-4">
        <div className="row">
            {products.map(product => (
                <div key={product.id} className="col-md-4">
                    <div className="card mb-4">
                        <div className="card-body">
                            <h5 className="card-title">{product.name} - ${product.price}</h5>
                            <p className="card-text">{product.description}</p>
                            <a href={`/products/${product.id}`} className="btn btn-primary">Details</a>
                        </div>
                    </div>
                </div>
            ))}
        </div>
    </div>
);
```

### 11. Add React Router for navigation:

Install react-router-dom:

```bash
npm install react-router-dom
```

Set up routes in `App.js`:

```jsx
import { BrowserRouter as Router, Route } from 'react-router-dom';
import ProductDetail from './ProductDetail';

function App() {
    return (
        <Router>
            <div className="App">
                <header className="App-header bg-primary text-white p-4">
                    <h1>Product Manager</h1>
                </header>
                <Route path="/" exact component={Products} />
                <Route path="/products/:id" component={ProductDetail} />
            </div>
        </Router>
    );
}
```

### 12. Add a "Create Product" Form:

Let's create a form to add new products:

`CreateProduct.js`:

```jsx
import React, { useState } from 'react';
import axios from 'axios';

function CreateProduct() {
    const [name, setName] = useState('');
    const [price, setPrice] = useState('');
    const [description, setDescription] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();

        const newProduct = { name, price: parseFloat(price), description };
        axios.post('http://localhost:3001/products', newProduct)
            .then(response => {
                if (response.status === 201) {
                    alert("Product created successfully!");
                    setName('');
                    setPrice('');
                    setDescription('');
                }
            });
    };

    return (
        <div className="container mt-4">
            <h2>Create Product</h2>
            <form onSubmit={handleSubmit}>
                <div className="form-group">
                    <label>Name</label>
                    <input type="text" className="form-control" value={name} onChange={e => setName(e.target.value)} required />
                </div>
                <div className="form-group">
                    <label>Price</label>
                    <input type="number" className="form-control" value={price} onChange={e => setPrice(e.target.value)} required />
                </div>
                <div className="form-group">
                    <label>Description</label>
                    <textarea className="form-control" value={description} onChange={e => setDescription(e.target.value)} required></textarea>
                </div>
                <button type="submit" className="btn btn-primary">Add Product</button>
            </form>
        </div>
    );
}

export default CreateProduct;
```

Add the new route in `App.js`:

```jsx
<Route path="/create" component={CreateProduct} />
```

### Reasoning:

- **Bootstrap**: Provides pre-styled components and classes that allow for quick, consistent UI design.
  
- **ProductDetail component**: Allows users to explore more about a product, enhancing UX.
  
- **React Router**: Necessary for navigating between different components/pages without a full page refresh, giving a SPA feel.
  
- **CreateProduct component**: Allows users to add new products, thereby demonstrating POST operations against the mock server.

Now, our application has a more comprehensive feel, with the ability to navigate to product details, and the functionality to add new products.

**Comment by Ruchiarth** [comment]
Part 1 
Error : 
![image](https://github.com/akash-coded/mern/assets/126975575/a7cf213c-12d0-48a5-838b-7c469611a19e)

Solution 
npm install -g npm-windows-upgrade
npm install -g concurrently

![image](https://github.com/akash-coded/mern/assets/126975575/5ad2a14b-a7be-4a40-ab69-26f53066e74a)

**Reply by akash-coded** [reply]
Good job

**Reply by Ruchiarth** [reply]
![image](https://github.com/akash-coded/mern/assets/126975575/494e71e3-7e8e-4e12-8cfd-5a483ff2deb9)
![image](https://github.com/akash-coded/mern/assets/126975575/7f49b9f5-0068-420d-9414-c7c3e0084bc4)
![image](https://github.com/akash-coded/mern/assets/126975575/7277d98e-91ab-4d9c-b584-344a2cfc9f83)
![image](https://github.com/akash-coded/mern/assets/126975575/3bafa30d-eaf4-42fa-98d9-ed35707a8fce)
error: 
![image](https://github.com/akash-coded/mern/assets/126975575/ad86c99f-f75c-4e6d-b552-c1f7f0b1590e)
Silly one 
Solution : import { useParams } from "react-router-dom";
correct spelling of Params
Also in Productdetails.js
useEffect(() => {
    axios
      .get(`http://localhost:3001/products/${id}`)
      .then((response) => {
        setProduct(response.data);
      })
      .catch((error) => {
        setError(error);
      });
  }, [id]);
//Mention ID 

Err1 : Routes under Route 
Sol 👍 in App.js 
```
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
return (
    <Router>
      <div className="App">
        <header className="App-header bg-primary text-white p-4">
          <h1>Product Manager</h1>
        </header>
        <Routes>
          <Route path="/" element={<Products />} />
          <Route path="/products/:id" element={<ProductDetail />} />
          <Route path="/create" element={<CreateProduct />} />
        </Routes>
      </div>
    </Router>
```

**Comment by bipin0108** [comment]
![Screenshot from 2023-08-12 15-55-08](https://github.com/akash-coded/mern/assets/14278619/f9eaa48c-8544-4cbd-b3f9-9dfcb42a9d4c)
![Screenshot from 2023-08-12 15-55-14](https://github.com/akash-coded/mern/assets/14278619/402504d5-0b64-46f8-a3c1-f664a84067f5)
![Screenshot from 2023-08-12 15-55-25](https://github.com/akash-coded/mern/assets/14278619/781bed84-15d0-4c1b-a7cc-a57657fb47df)

**Reply by akash-coded** [reply]
Perfect. Please showcase with more products and also with how you fixed errors, if any

**Reply by bipin0108** [reply]
I got 2 or 3 issues and I have resolved using searching the internet.

1. concurrently npm
2. missing the routes in browser-router and route 
3.  and not show the component after the implementing router dome
4.  usePrams not desfine in details product

**Comment by mownica-p** [comment]
![image](https://github.com/akash-coded/mern/assets/126975660/b7b2e013-6f7d-4c54-87d5-b9d0cba71190)
npm install -g concurrently

![image](https://github.com/akash-coded/mern/assets/126975660/a5a97485-5384-4480-b5ef-b67d318e98e8)

![image](https://github.com/akash-coded/mern/assets/126975660/5131f7df-e413-4b1f-95bb-b6259bc4fb0f)

![image](https://github.com/akash-coded/mern/assets/126975660/b29db2ea-2fa4-4056-a850-d1d9d766c40b)

![image](https://github.com/akash-coded/mern/assets/126975660/387a8679-9f56-409f-af4d-6750efe1220f)

![image](https://github.com/akash-coded/mern/assets/126975660/23c3ecb4-e3f9-41df-af1f-944e357d8180)

![image](https://github.com/akash-coded/mern/assets/126975660/c7bbb75f-722c-42ce-aaeb-4a536287feac)

![image](https://github.com/akash-coded/mern/assets/126975660/7e535e56-504d-46dc-b7b8-e358ad39106d)

![image](https://github.com/akash-coded/mern/assets/126975660/fab56ecf-0d24-40c6-8833-26f4820e7bc5)

**Comment by sam2elect** [comment]
# Resolving Errors


![Screenshot 2023-08-12 195629](https://github.com/akash-coded/mern/assets/85131631/e1867deb-61cb-4c78-a164-f294564810e7)
```js
npm install concurrently --save-dev
```
![Screenshot 2023-08-12 195807](https://github.com/akash-coded/mern/assets/85131631/18b6a2ba-17f7-48d8-8d7e-38983eed72e3)
![Screenshot 2023-08-12 200347](https://github.com/akash-coded/mern/assets/85131631/46edba09-b4d3-4c40-b844-5a9697d5722d)
```js
change product in the use state to products
  const [products, setProducts] = useState([]);
```
![Screenshot 2023-08-12 200549](https://github.com/akash-coded/mern/assets/85131631/ad1822c0-6d1f-428e-ac61-e6aa6e4a7bc2)

```js
//import Create product from ./createproduct
import CreateProduct from "./CreateProduct";
```
![Screenshot 2023-08-12 200715](https://github.com/akash-coded/mern/assets/85131631/3fa022bb-c304-49f3-9621-80b1afb59319)
```js
add routes  import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
<Routes>
          <Route path="/" element={<Products />} />
          <Route path="/products/:id" element={<ProductDetail />} />
          <Route path="/create" element={<CreateProduct />} />
        </Routes>
```
![image](https://github.com/akash-coded/mern/assets/85131631/dd37777a-3bdd-48ab-b7e5-37dbb157ab05)
![image](https://github.com/akash-coded/mern/assets/85131631/86b2d0e9-b599-47e9-a12b-7e483bb54957)
![image](https://github.com/akash-coded/mern/assets/85131631/e820f691-3a33-45cc-9d99-cfdcc4350369)
![image](https://github.com/akash-coded/mern/assets/85131631/2ca20144-7e92-41d2-a3d8-9b7d86dc0bec)
![image](https://github.com/akash-coded/mern/assets/85131631/423ab55d-b716-4767-af8a-21cd103b463a)
![image](https://github.com/akash-coded/mern/assets/85131631/45c23321-7910-4cfd-92bf-bbcd574a4f95)


```js

product.js

import { Link, useNavigate } from "react-router-dom"; // Import useNavigate
  // Function to navigate to ProductDetail
  const viewProduct = (productId) => {
    navigate(`/products/${productId}`);
  };
<h3 onClick={() => viewProduct(product.id)}>
            {product.name} - ${product.price}
          </h3>

productDetail.js

import { useParams, useNavigate } from "react-router-dom";
 //import use params 
  const { id } = useParams();
remove match prop 
ProductDetail() 
Remove the   {products.map(product => ())
as there is only one product 

change the anchor to button and add onclick function
 <button         
                className="btn btn-primary"
                onClick={navigateToCreateProduct}
               > 
               Details
              </button>

createproduct.js
set the state variables from empty top take user input
if (response.status === 201) {
          alert("Product created successfully!");
          setName(name);
          setPrice(price);
          setDescription(description);
        }
      });

App.js
Remove the AppHeader className from header tag
        <header className="bg-primary text-white p-4">

```

**Comment by starvikas1988** [comment]
you need to enable javascript to
 run this app. react
<img width="781" alt="image" src="https://github.com/akash-coded/mern/assets/70792693/d44e746f-c367-491c-844c-c0ab3274226b">

![err3](https://github.com/akash-coded/mern/assets/70792693/8b678020-b7df-4123-97af-7fc4b1a2e7b7)
![err4](https://github.com/akash-coded/mern/assets/70792693/20705f05-3cee-4182-9259-182ea839905d)
![products](https://github.com/akash-coded/mern/assets/70792693/2cda9b3d-9a1d-4702-802b-d87d16bff7e2)
![product_details](https://github.com/akash-coded/mern/assets/70792693/be32b3f5-30b7-4ce8-88a1-2f6c4e54464b)

# Products.js
```js
import React, { useState, useEffect } from 'react';
import axios from 'axios';

function Products() {
  const [products, setProducts] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    axios.get('http://localhost:3001/products')
      .then(response => {
        setProducts(response.data);
      })
      .catch(error => {
        setError(error);
      });
  }, []);

  if (error) return <div>Error fetching products!</div>;

  return (
    <div className="container mt-4">
        <div className="row">
            {products.map(product => (
                <div key={product.id} className="col-md-4">
                    <div className="card mb-4">
                        <div className="card-body">
                            <h5 className="card-title">{product.name} - ${product.price}</h5>
                            <p className="card-text">{product.description}</p>
                            <a href={`/products/${product.id}`} className="btn btn-primary">Details</a>
                        </div>
                    </div>
                </div>
            ))}
        </div>
    </div>
);
}

export default Products;
```
# ProductDetail.js

```js
import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { useParams } from "react-router-dom";

function ProductDetail({ match }) {
    const [product, setProduct] = useState({});
    const [error, setError] = useState(null);
    const { id } = useParams();
   

    useEffect(() => {
        axios.get(`http://localhost:3001/products/${id}`)
            .then(response => {
                setProduct(response.data);
            })
            .catch(error => {
                setError(error);
            });
    }, [id]);

    if (error) return <div>Error fetching product details!</div>;

    return (
        <div className="container mt-4">
            <h2>{product.name}</h2>
            <h4>${product.price}</h4>
            <p>{product.description}</p>
        </div>
    );
}

export default ProductDetail;
```
# App.js
```js
import React from 'react';
import { BrowserRouter as Router,Routes, Route } from 'react-router-dom';

import './App.css';
import Products from './Products';
import ProductDetail from './ProductDetail';
import CreateProduct from './CreateProduct';

function App() {
  return (
    <Router>
        <div className="App">
      <header className="App-header bg-primary text-white p-4">
          <h1>Product Manager</h1>
      </header>
<Routes>
      <Route path="/" exact element ={<Products/>} />
      <Route path="/products/:id" element ={<ProductDetail/>} />
      <Route path="/create" element ={<CreateProduct/>} />
</Routes>
</div>

    </Router>
);

}


export default App;
```

# index.js
```js
import React from 'react';
import ReactDOM from 'react-dom/client';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';



const root = ReactDOM.createRoot(document.getElementById('root'));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();

```

**Reply by starvikas1988** [reply]
<img width="569" alt="image" src="https://github.com/akash-coded/mern/assets/70792693/968abf03-d87c-4fdd-9856-7ad900a5c941">

**Comment by tinapatil13** [comment]
Part1:
![Screenshot 2023-08-13 172448](https://github.com/akash-coded/mern/assets/88702583/8a47c6cf-d741-4abf-9cae-409a25e62949)
![part2half](https://github.com/akash-coded/mern/assets/88702583/4839aee0-076e-4f46-81ed-8798fbb47df9)
Note : facing issue of upperCase so in that case restart your tool and  after using npm run dev command I got concurrently not recognize in that case use npm install concurrently --save-dev. 
Part 2:
![image](https://github.com/akash-coded/mern/assets/88702583/501c7990-cd2d-4573-8a66-18e08bb23b93)
![image](https://github.com/akash-coded/mern/assets/88702583/bb76123f-5c51-4e7b-8beb-4a65c7ad8093)
Note: In 2nd part App.js  route never wrap up directly it's child element of routes so use: 
 import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
<Routes>
          <Route path="/" element={<Products />} />
          <Route path="/products/:id" element={<ProductDetail />} />
          <Route path="/create" element={<CreateProduct />} />
 </Routes>

**Comment by pujita789** [comment]
#  App.js
import React from 'react';
import './App.css';
import Products from './products';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import ProductDetail from './productDetail';
import CreateProduct from './CreateProduct';

function App() {
    return (
        <Router>
            <div className="App">
                <header className="App-header bg-primary text-white p-4">
                    Product Manager
                </header>
                <Routes>
                    <Route path="/" exact element={<Products />} />
                <Route path="/products/:id" component={ProductDetail} />
                <Route path='/create' component={CreateProduct}/>
                </Routes>
                
            </div>
        </Router>
    );
}

export default App;
#  Products.js
import React, { useState, useEffect } from 'react';
import axios from 'axios';

function Products() {
  const [products, setProducts] = useState([]);
  const [error, setError] = useState(null);

  useEffect(() => {
    axios.get('http://localhost:3001/products')
      .then(response => {
        setProducts(response.data);
      })
      .catch(error => {
        setError(error);
      });
  }, []);

  if (error) return <div>Error fetching products!</div>;

  return (
    <div className="container mt-4">
        <div className="row">
            {products.map(product => (
                <div key={product.id} className="col-md-4">
                    <div className="card mb-4">
                        <div className="card-body">
                            <h5 className="card-title">{product.name} - ${product.price}</h5>
                            <p className="card-text">{product.description}</p>
                            <a href={`/products/${product.id}`} className="btn btn-primary">Details</a>
                        </div>
                    </div>
                </div>
            ))}
        </div>
    </div>
);
}

export default Products;

#   Productdetails.js

import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { useParams } from "react-router-dom";

function ProductDetail({ match }) {
    const [product, setProduct] = useState({});
    const [error, setError] = useState(null);
    const { id } = useParams();
   

    useEffect(() => {
        axios.get(`http://localhost:3001/products/${id}`)
            .then(response => {
                setProduct(response.data);
            })
            .catch(error => {
                setError(error);
            });
    }, [id]);

    if (error) return <div>Error fetching product details!</div>;

    return (
        <div className="container mt-4">
            <h2>{product.name}</h2>
            <h4>${product.price}</h4>
            <p>{product.description}</p>
        </div>
    );
}

export default ProductDetail;

#  CreateProduct.js

import React, { useState } from 'react';
import axios from 'axios';

function CreateProduct() {
    const [name, setName] = useState('');
    const [price, setPrice] = useState('');
    const [description, setDescription] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();

        const newProduct = { name, price: parseFloat(price), description };
        axios.post('http://localhost:3001/products', newProduct)
            .then(response => {
                if (response.status === 201) {
                    alert("Product created successfully!");
                    setName('');
                    setPrice('');
                    setDescription('');
                }
            });
    };

    return (
        <div className="container mt-4">
            <h2>Create Product</h2>
            <form onSubmit={handleSubmit}>
                <div className="form-group">
                    <label>Name</label>
                    <input type="text" className="form-control" value={name} onChange={e => setName(e.target.value)} required />
                </div>
                <div className="form-group">
                    <label>Price</label>
                    <input type="number" className="form-control" value={price} onChange={e => setPrice(e.target.value)} required />
                </div>
                <div className="form-group">
                    <label>Description</label>
                    <textarea className="form-control" value={description} onChange={e => setDescription(e.target.value)} required></textarea>
                </div>
                <button type="submit" className="btn btn-primary">Add Product</button>
            </form>
        </div>
    );
}

export default CreateProduct;

# db.json

{
  "products": [
    {
      "id": 1,
      "name": "Laptop",
      "price": 1000,
      "description": "A high performance laptop"
    },
    {
      "id": 2,
      "name": "Mouse",
      "price": 20,
      "description": "Ergonomic wireless mouse"
    },
    {
      "id": 3,
      "name": "Earpods",
      "price": 120,
      "description": "Ergonomic wireless earpods, with good earbase."
    },
    {
      "id": 4,
      "name": "BackPack",
      "price": 300,
      "description": "Backpacks with waterproof layer."
    }
  ]
}





![pmproduct](https://github.com/akash-coded/mern/assets/117858130/80aafb6d-fd9e-4999-9913-377867272677)

**Comment by tinapatil13** [comment]
[Discussion187.zip](https://github.com/akash-coded/mern/files/12329723/Discussion187.zip)

**Comment by vk0027** [comment]
Part 1: 
error :
![image](https://github.com/akash-coded/mern/assets/123943655/0a329965-2bd8-45a7-816c-6c1e0b48a4e3)

solution :
npm install -g npm-windows-upgrade
npm install -g concurrently

![image](https://github.com/akash-coded/mern/assets/123943655/e4e11c27-efad-4d6a-8f75-5bcfd1f08b4c)

![image](https://github.com/akash-coded/mern/assets/123943655/4944aeb9-5fc5-47bc-9828-b37a222e15f2)

**Comment by RevathiSundaramoorthy** [comment]
1.	Error on concurrently - npm install -g concurrently
2.	Not routing to child components – changed Router to routes and component to element
3.	Used UseParams and ${id}

![image](https://github.com/akash-coded/mern/assets/126975777/d8becff5-969e-4de9-86c7-904aca92778b)
![image](https://github.com/akash-coded/mern/assets/126975777/a057e588-63dd-4f2f-ab08-6ad4b8a0f1ae)
![image](https://github.com/akash-coded/mern/assets/126975777/6d8304ac-3514-4e03-bc29-0788707883f7)
![image](https://github.com/akash-coded/mern/assets/126975777/9630ee72-d50d-4b18-9df3-3d604efb3f2d)
![image](https://github.com/akash-coded/mern/assets/126975777/96a78b60-baa4-4845-b174-78b3b1e3280d)
